pages/advanced/api-safe-transaction-service/guides/api.md

```diff
@@ -0,0 +1,627 @@
+  const apiKit = new SafeApiKit({
+    chainId: 11155111n
+  })
+
+  // Get signed Tx
+  const signedTransaction = await apiKit.getTransaction(safeTxHash)
+
+  // Log Tx confirmations
+  const confirmations = signedTransaction.confirmations
+  console.log('Tx hash %s confirmations: %O', safeTxHash, confirmations)
+
+  // Instantiate an EthAdapter for Owner A
+  const ethProvider = new ethers.JsonRpcProvider(config.RPC_URL)
+
+  const ownerA = new ethers.Wallet(config.OWNER_A_PRIVATE_KEY, ethProvider)
+  const ethAdapterOwnerA = new EthersAdapter({
+    ethers,
+    signerOrProvider: ownerA
+  })
+
+  // Initialize the Protocol Kit for Owner A
+  const protocolKitOwnerA = await Safe.create({
+    ethAdapter: ethAdapterOwnerA,
+    safeAddress: config.SAFE_ADDRESS
+  })
+
+  // Execute Tx
+  const transactionResponse =
+    await protocolKitOwnerA.executeTransaction(signedTransaction)
+  console.log('Executed Tx: %O', transactionResponse)
+}
+```
+
+#### Using safe-eth-py
+
+```python
+from gnosis.eth import EthereumClient, EthereumNetwork
+from gnosis.safe.api.transaction_service_api import TransactionServiceApi
+
+def get_and_execute_signed_tx(safe_tx_hash):
+    # Instantiate a Tx Service Api
+    ethereum_client = EthereumClient(config.get("RPC_URL"))
+
+    transaction_service_api = TransactionServiceApi(
+        network=EthereumNetwork.SEPOLIA,
+        ethereum_client=ethereum_client
+    )
+
+    # Get Tx from Tx Service
+    (safe_tx_from_tx_service, _) = transaction_service_api.get_safe_transaction(safe_tx_hash)
+
+    # Execute Tx
+    result = safe_tx_from_tx_service.execute(config.get("OWNER_A_PRIVATE_KEY"))
+    print(f"Tx executed: {result}")
+```
+
+### Getting the last executed transaction
+
+#### Using Curl
+
+```bash
+curl -X 'GET' \
+'https://safe-transaction-sepolia.safe.global/api/v1/safes/0xc62C5cbB96ffffffffffff2f78A4d3071317ffff/multisig-transactions/?executed=true&limit=1' \
+-H 'accept: application/json' \
+```
+
+#### Using safe-core-sdk
+
+```typescript
+import SafeApiKit from '@safe-global/api-kit'
+
+async function getLastExecutedTx() {
+  const apiKit = new SafeApiKit({
+    chainId: 11155111n
+  })
+
+  const transactions = await apiKit.getMultisigTransactions(config.SAFE_ADDRESS)
+
+  if (transactions.results.length > 0) {
+    console.log('Las executed TX: %O', transactions.results[0])
+    return
+  }
+
+  console.log('No Tx found')
+}
+```
+
+#### Using safe-eth-py
+
+```python
+from gnosis.eth import EthereumClient, EthereumNetwork
+from gnosis.safe.api.transaction_service_api import TransactionServiceApi
+
+def get_last_executed_tx():
+    # Instantiate a Tx Service Api
+    ethereum_client = EthereumClient(config.get("RPC_URL"))
+    transaction_service_api = TransactionServiceApi(
+        network=EthereumNetwork.SEPOLIA,
+        ethereum_client=ethereum_client
+    )
+
+    # Get Tx from Tx Service
+    transactions = transaction_service_api.get_transactions(config.get("SAFE_ADDRESS"))
```

Comment:
Don't we allow query params in our API? We should refactor `safe-eth-py` to allow them

Reply:
It seems that no: https://github.com/safe-global/safe-eth-py/blob/93635053697a57849920e86821bc6eec69967412/gnosis/safe/api/transaction_service_api/transaction_service_api.py#L159

Reply:
Could you create an issue on safe-eth-py repo?

Reply:
Done: https://github.com/safe-global/safe-eth-py/issues/890